WordPress/WordPressShareExtension/ShareModularViewController.swift

```diff
@@ -808,8 +814,33 @@ fileprivate extension ShareModularViewController {
 
         // Then proceed uploading the actual post
         let networkService = AppExtensionsService()
+        let isAuthorizedToUploadFiles = networkService.isAuthorizedToUploadMedia(in: siteList, for: siteID)
         let localImageURLs = [URL](shareData.sharedImageDict.keys)
-        if !localImageURLs.isEmpty {
+
+        if localImageURLs.isEmpty {
```

Comment:
Would it make sense to encapsulate each branch of the `if else` in a separate method? It might make this method a bit more readable, what do you think?

Reply:
+1 to @ctarda's comment ☝️ . Then the `isAuthorizedToUploadMedia` check could be at the top of the new "uploadPostWithMedia" func...perhaps as `guard` statement e.g.:

```
let isAuthorizedToUploadFiles = networkService.isAuthorizedToUploadMedia(in: siteList, for: siteID)
 guard isAuthorizedToUploadFiles else {
            // Error: this role is unable to upload media.
            let error = self.createErrorWithDescription("This role is unable to upload media.")
            self.tracks.trackExtensionError(error)
            let title = NSLocalizedString("Sharing Error", comment: "Share extension error dialog title.")
            let message = NSLocalizedString("This role is unable to upload media. Please contact your Site Administrator.", comment: "Share extension error dialog text.")
            self.showAlert(title: title, message: message, retry: false)
            return
        }
```